### PR TITLE
clean git implementation slightly

### PR DESCRIPTION
**Pull request by biox**

### CODE REVIEW

**Comment by biox** [comment]
tried to make it a little more obvious - felt wrong that the function `git_add_and_commit` would perform any checking about whether git was available - i moved those checks to the relevant callsites.

also, i busted the add & commit steps apart and added naive error handling. idk if that's ideal but i figure if automated git stuff borks we should probably toss an error of our own.

File: pa [hunk 1]
```diff
@@ -99,7 +99,8 @@ pw_list() {
 }
 
 git_add_and_commit() {
-    if $git_enabled; then git add "$1" && git commit -qm "$2"; fi
+    git add "$1" || die "failed to git add $1"
```
Comment (arcxio) [inline pa:102]: style nits:

- "failed to" -> "Couldn't"
- `die` should be on its own line
- there should be an empty line in-between add and commit

Comment (biox) [reply pa]: nits addressed in 861d1c9

File: pa [hunk 2]
```diff
@@ -99,7 +99,11 @@ pw_list() {
 }
 
 git_add_and_commit() {
-    if $git_enabled; then git add "$1" && git commit -qm "$2"; fi
+    git add "$1" ||
+        die "couldn't git add $1"
```
Comment (arcxio) [inline pa:103]: all of the other error messages are Capitalized (except when it starts with the name of the utility like `age`).
I'm not against lowercase errors - in fact I think it's better for aligning with unix convention - but I believe it has to be consistent across the project.

Comment (biox) [reply pa:103]: ok :D https://github.com/biox/pa/pull/45/commits/d575d81a8883b98ecebb189d741d78a45448ae4e

Comment (arcxio) [reply pa:103]: radical, I respect that. there are contribs left, but they can be handled separately

**Review by arcxio — APPROVED** [review]

**Pull request merged**